### PR TITLE
RUMM-1346 Add nightly tests for Logger.Builder and Datadog.initialize

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

In this PR we are adding the nightly test cases for the `Logger#Builder` options and also for the `Datadog#initialize` configuration settings related with the Logs feature.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            logger = Logger.Builder().setSampleRate(1f).build()
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+
+    /**
+     * apiMethodSignature: Logger#Builder#fun setSampleRate(Float): Builder
+     */
+    @Test
+    fun logs_logger_builder_sample_all_out() {
```
Comment (xgouchet) [inline instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt:56]: Maybe we need one or more to test what happens when we use a value between 0 and 1, e.g 75%

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        measureLoggerInitialize {
+            logger = Logger.Builder().setBundleWithRumEnabled(false).build()
+        }
+        val viewKey = forge.anAlphabeticalString()
+        GlobalRum.get().startView(viewKey, forge.anAlphabeticalString())
+        logger.sendRandomLog(testMethodName, forge)
+        GlobalRum.get().stopView(viewKey)
+    }
+
+    companion object {
```
Comment (xgouchet) [inline instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt:166]: You're missing test for `setBundleWithTraceEnabled`

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsConfigE2ETests.kt [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                        }
+                    }
+                )
+                    .build()
+            )
+        }
+        measureLoggerInitialize {
+            logger = initializeLogger()
+        }
+        logger.sendRandomLog(testMethodName, forge)
```
Comment (xgouchet) [inline instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LogsConfigE2ETests.kt:148]: You should measure this too

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt (hunk 1 above)
Comment (mariusc83) [reply instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt:56]: Yes, I also thought of that but then problem is how would you assert that test case as you are not sure if this event is passing your sampling or not ?

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt (hunk 2 above)
Comment (mariusc83) [reply instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt:166]: Nope, as I explained last week during the daily I could not find a way for now to assert this test case. The `dd.trace_id` or `dd.span_id` are not queryable and not even visible in the Log event attributes dashboard.

Comment (xgouchet) [reply instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt:166]: Even though you can't assert that the `dd.trace_id` and `dd.span_id`  are present, you can still assert that the log exist, right? A partial assertion is always better than no assertions at all

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt (hunk 1 above)
Comment (xgouchet) [reply instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt:56]: You send `n` logs and assert that the number of logs with the matching tag you receive is roughly equal to `n⨉75%`

**Review by 0xnm — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]

**Pull request merged**